File: domain/src/main/kotlin/com/jeferro/products/domain/authentications/entities/UserCredentials.kt
package com.jeferro.products.domain.authentications.entities

import com.jeferro.products.domain.shared.entities.entities.Aggregate
import com.jeferro.products.domain.shared.entities.metadata.Metadata
import com.jeferro.products.domain.shared.entities.auth.UserId
import com.jeferro.products.domain.shared.entities.auth.Username

class UserCredentials(
    id: UserId,
    val username: Username,
    val encodedPassword: String,
    val roles: List<String>,
    metadata: Metadata
) : Aggregate<UserId>(id, metadata)

File: .docker/mongo/initdb/user-credentials.js
let now = new Date()

db['user-credentials'].insertMany([
    {
        _id: "user",
        username: "user",
        encodedPassword: "$2a$10$wWY9tn/jnn66d6gsBpGM3.KZXXn7AmN0U/JVGzioJjtQFOJkCYFVa",
        roles: [
            'user'
        ],
        metadata: {
            createdAt: now,
            createdBy: 'user',
            updatedAt: now,
            updatedBy: 'user'
        },
        _class: 'com.jeferro.products.components.products.mongo.authentications.dtos.UserCredentialsMongoDTO'
    }
])
